### DISCUSSION TITLE
"Why is the compression efficiency of avifenc inferior to that of the third-party conversion software iYo Image Converter?

### DISCUSSION THREAD

**Discussion by sojuo** [AOMediaCodec/libavif#2533] [Q&A]
In order to obtain an AVIF image where every pixel displayed on the screen matches the original image, I used the command `avifenc --lossless -j all -y 444 --depth 12 --qcolor 100 --qalpha 100 --min 0 --max 0 --minalpha 0 --maxalpha 0 --speed 0 * *.avif` and [iYo Image Converter](https://github.com/ssnangua/iyo-image-converter) to compress PNG images. Surprisingly, avifenc was less efficient than iYo Image Converter, with avifenc's compressed images occupying more storage space than those compressed by [iYo Image Converter](https://github.com/ssnangua/iyo-image-converter).
Original Image:![original](https://github.com/user-attachments/assets/0c87dd98-2fcd-4f36-a724-e54b7d3f83d0)

Images generated by the avifenc command(33.9 KB):![avifenc](https://files.catbox.moe/j43vk1.avif)

Images converted by [iYo Image Converter](https://github.com/ssnangua/iyo-image-converter)(23.7 KB):![iYo-Image-Converter](https://files.catbox.moe/nplqnq.avif)

[iYo Image Converter](https://github.com/ssnangua/iyo-image-converter) Settings:![Settings](https://files.catbox.moe/ayrz27.avif)

My goal is to obtain a lossless AVIF image where every pixel displayed on the screen matches the original image. I would like to know if the issue lies with the parameters I entered for avifenc, or if there is another reason why the images generated by avifenc are not the smallest. Is there a way to make avifenc generate the smallest possible AVIF image?

Additionally, you can find all the files I generated here:[files.zip](https://github.com/user-attachments/files/18275516/files.zip)

**Comment by y-guyon** [comment]
Thank you for your interest in libavif.

It seems the AVIF files in your `files.zip` archive do not decode to exactly the same pixels as `original.png` when converted with `avifdec`. I used the [imagemagick compare tool](https://imagemagick.org/script/compare.php) to check for PNG equivalence.

Please try the following command line:

```sh
avifenc --lossless input.png output.avif
```

**Reply by sojuo** [reply]
> Thank you for your interest in libavif.
> 
> It seems the AVIF files in your `files.zip` archive do not decode to exactly the same pixels as `original.png` when converted with `avifdec`. I used the [imagemagick compare tool](https://imagemagick.org/script/compare.php) to check for PNG equivalence.
> 
> Please try the following command line:
> 
> ```shell
> avifenc --lossless input.png output.avif
> ```

Thank you for the command. It generated a 32KB file, but it is still larger than the 24KB file produced by iYo Image Converter.

**Reply by y-guyon** [reply]
But is the 24KB file produced by iYo Image Converter mathematically lossless? It was not the case in my local test based on your `files.zip`.

If you are only interested in visually lossless and not mathematically lossless, I would suggest some high quality instead of `--lossless`:

```sh
avifenc -q 90 --speed 0 input.png output.avif
```

**Reply by sojuo** [reply]
> But is the 24KB file produced by iYo Image Converter mathematically lossless? It was not the case in my local test based on your `files.zip`.
> 
> If you are only interested in visually lossless and not mathematically lossless, I would suggest some high quality instead of `--lossless`:
> 
> ```shell
> avifenc -q 90 --speed 0 input.png output.avif
> ```

How can I determine if this image has been converted from the original losslessly?

**Reply by y-guyon** [reply]
As I mentioned in https://github.com/AOMediaCodec/libavif/discussions/2533#discussioncomment-11697782, I used `avifdec` to get back to PNG format, then an ImageMagick tool to highlight the differences:

```sh
avifdec iYO_Image_Converter.avif iYO_Image_Converter.avif.png
compare original.png iYO_Image_Converter.avif.png diff.png
```
`diff.png`
![diff](https://github.com/user-attachments/assets/edc9064e-d3e9-40d2-b2b7-98900f95c936)

To doublecheck with GIMP, the bottom status bar background color goes from `#073642` in `original.png` to `#073542` in `iYO_Image_Converter.avif` for example.
